### ISSUE TITLE
fix(training): RAM reward re-renders each target once per group member

### ISSUE THREAD

**Issue by ktinubu** [tinaudio/synth-setter#3254]
`VSTFlowRAMModule.training_step` (`src/synth_setter/models/vst_flow_ram_module.py`) passes `batch[target_key].repeat_interleave(group)` to the reward, so `SynthRenderedReward.forward` (`src/synth_setter/models/components/rendered_reward.py`) re-renders each identical target row once per group member; every render also goes through the synth initialization and preset-digest path.

**What:** with `experiment=surge/flow_ram_simple` (`batch_size=8`, `num_samples_per_row=8`) each step performs 128 surgepy renders of which 56 target renders are exact duplicates; roughly 44% of the per-step render budget is redundant.
**Expected:** render each unique target once and broadcast the audio across its group (or cache the target audio per row). The reward values and RAM loss are unchanged.

Refs #3150 (deferred from https://github.com/tinaudio/synth-setter/pull/3151 review).

**Comment by ktinubu** [comment]
The redundant target-render portion is now addressed in PR #3151 by `f96b65ccd1f261124634d7227c49882e3be4d68f`: each original conditioning row is rendered once before candidate grouping, and that waveform is reused across its `G` candidates. Candidate rows still render independently, so this does not claim to eliminate Surge phase randomness.

Local verification passed all 42 focused reward/RAM tests, including `B + B*G` render-count, row-alignment, grouping, and K-preservation coverage, plus real Surge/surgepy and TorchSynth pretrain → RAM training E2Es.

Keeping this issue open: the controlled-render-randomness assessment and real-Surge before/after reward-variance characterization remain research acceptance scope. The PR description now states that narrower contract rather than claiming the shared render path sidesteps phase randomisation.

**Comment by ktinubu** [comment]
### [from W7] The code scope already shipped **in #3151 itself** — and the measurement shows target reuse accounts for only ~18% of the noise

Claimed this as unclaimed (no assignee, branch, worktree or PR) and worked it end to end. It needs no PR. Evidence below, run against real Surge on `main` (`84da89baf3`).

## The target is already rendered once per original row

`VSTFlowRAMModule.training_step` renders the target **before** grouping and only then repeats it:

```python
raw_target = batch[self._reward_target_key]
prepare_target = getattr(self.reward, "prepare_target", None)
target = cast(Tensor, prepare_target(raw_target)) if callable(prepare_target) else ...
target = target.repeat_interleave(group, dim=0)   # one render, shared by the group
```

`git log -S "prepare_target"` gives exactly one commit: **`6326f53dec` — #3151**. So this landed in the very PR the issue was deferred from; the deferral was recorded against an earlier state of that PR and its final state already satisfies it. `SynthRenderedReward.prepare_target` / `target_key` exist for this purpose, and `tests/models/test_synth_rendered_reward.py::test_synth_rendered_reward_reuses_one_target_render_per_original_row` already pins it (6 passed).

That covers acceptance criteria 1, 2 and 5.

## AC 4 has no target text

The doc line the scope names — sharing the render path *"sidesteps Surge's per-render phase randomisation"* — is **not on `main`**. `git grep -inE "sidestep|phase randomi" origin/main` returns only unrelated CI comments and CHANGELOG history. Nothing to correct.

## AC 3, measured — and it does not say what the issue expects

Real surgepy in-process renders, `surge_simple_surgepy`, 16 kHz, 1 s, the production `MultichannelAudioDistance` (`spectral 1.0 / channel_mldr 0.1 / pair_mldr 0.1`). 12 groups × group size 8, with all 8 candidate rows set **identical to the target row**, so every non-zero spread is pure render noise:

| | mean | median | min | max |
| --- | --- | --- | --- | --- |
| within-group spread, target rendered **once** (current) | **1.038151** | 1.131903 | 0.093595 | 2.206322 |
| within-group spread, target **re-rendered per candidate** (pre-#3151) | **1.266841** | 1.127222 | 0.108808 | 3.313668 |
| reward of a render scored against **itself** | **0.000000** | -0.000000 | -0.000000 | -0.000000 |

**Target reuse reduces mean within-group spread by 18.1%, not to zero.** The residual `1.038` is entirely *candidate* re-rendering — which this issue's scope explicitly keeps unchanged ("Keep candidate rendering and sampler behavior unchanged").

Two things follow, and they matter for how this issue reads:

1. **The prior reviewer's observation is reproduced but re-attributed.** Their spread of `1.515486` from eight identical rows sits inside my pre-#3151 range (0.109–3.314, mean 1.267) — so the number was right, but it is *not* mostly caused by re-rendering the target. Roughly four fifths of it survives target reuse. Reading that observation as "fixing the target render fixes the noise" would have been wrong.
2. **The distance module is exact**, so none of the spread is metric instability: a render scored against itself gives precisely `0.000000`. Two renders of *identical* parameters differ by `max|diff| = 0.004862` in the waveform — Surge randomises oscillator phase per render, as [the render-noise floor already established](https://github.com/tinaudio/synth-setter/issues/3150). The floor for a paired-randomness scheme is therefore 0, and the entire gap to it is renderer-side phase control.

## On "assess whether paired or otherwise controlled render randomness is useful"

Answered by the table: it is where the remaining ~82% lives, and it is **not reachable from the reward module** — it needs the Surge renderer to accept a phase seed, so every candidate and its target render on the same phase. That is a renderer capability, not a reward change, and it is a Feature rather than this Bug. I have not filed it, because the issue itself records that stochastic rewards stay mathematically valid for RL and that no gradient bias was established — so the case for spending renderer work on it should be made on measured training outcomes, not on this spread. Happy to file it if an owner wants it tracked.

## Recommendation

**Close as completed by #3151**, with AC 3 satisfied by the table above and AC 4 void. I am not closing it myself — the remaining scope item is a judgement call about renderer work, so I am leaving that to an owner rather than deciding it from the on-call queue.

Reproduce with `tests/models/test_synth_rendered_reward.py`'s own `_reward()` / `_rows()` helpers; no fixtures or GPU needed, ~2 min on CPU.
